app/controllers/api/v0/persons_controller.rb

```diff
@@ -17,8 +17,12 @@ def index
   def show
     person = Person.current.includes(:user, :ranksSingle, :ranksAverage).find_by_wca_id!(params[:wca_id])
     private_attributes = []
-    if current_user && current_user.can_admin_results?
-      private_attributes = %w[incorrect_wca_id_claim_count dob]
+    if current_user
+      if current_user.wca_id == params[:wca_id]
+        private_attributes = %w[dob]
+      elsif current_user.can_admin_results? || current_user.any_kind_of_delegate?
+        private_attributes = %w[incorrect_wca_id_claim_count dob]
+      end
```

Comment:
LGTM, but actually even delegate doesn't actually need `incorrect_wca_id_claim_count`.

Also how about storing `params[:wca_id]` to a variable as it is used more than once?

Also I was thinking to remove the nested if with following:

```
if current_user&.wca_id == wca_id || current_user&.any_kind_of_delegate?
  private_attributes = %w[dob]
elsif current_user&.can_admin_results?
  private_attributes = %w[incorrect_wca_id_claim_count dob]
end
```

But I am not sure which is better - @gregorbg might be able to help.